libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/DefaultTracer.java

```diff
@@ -85,6 +85,11 @@ public SpanScope withSpanInScope(Span span) {
         return DefaultSpanScope.create(span, tracerContextStorage).attach();
     }
 
+    @Override
+    public boolean isRecording() {
+        return true;
```

Comment:
```suggestion
        return tracingTelemetry.isTracingEnabled();
```

Reply:
tracingTelemetry is not aware of whether tracing is enabled or not. Its only TelemetrySettings which is not available in libs. Since, this is a static setting then we can pass a flag to the DefaultTracer through TracerFactory 
to make it aware of this setting?

Reply:
Ideally, if it is reaching till DefaultTracer the isRecording should be true.

Reply:
> tracingTelemetry is not aware of whether tracing is enabled or no

Ah you are right, I confused it with `TelemetrySettings`, apologizes for that